src/stream/src/executor/managed_state/top_n/top_n_state_new.rs

```diff
@@ -164,27 +164,26 @@ impl<S: StateStore> ManagedTopNStateNew<S> {
             }
         }
 
-        // fetch end row
-        if limit == 0 {
-            return Ok((start_row, invalid_row.clone()));
+        let mut valid_rows = vec![];
+        if start_row.is_valid() {
+            valid_rows.push(start_row.clone());
         }
-        let mut end_row = start_row.clone();
-        let num_limit = limit - 1;
-        for i in 0..num_limit {
+
+        // scan to the row at offset + limit
+        for _ in 0..(num_limit - 1) {
```

Comment:
I think this part looks a little bit tedious to me. Can we simply do something like:

```
state_table_iter.skip(offset).take(limit).map(|x| self.get_topn_row(x)).collect_vec().await?
```